Solaris doesn't have pretend, or pytest-benchmark packages,
hence all tests/options depending on those are disabled.

--- cryptography-50.0.0/pyproject.toml
+++ cryptography-50.0.0/pyproject.toml
@@ -140,7 +140,7 @@ exclude = [
 ]
 
 [tool.pytest.ini_options]
-addopts = "-r s --capture=no --strict-markers --benchmark-disable"
+addopts = "-r s --capture=no --strict-markers"
 testpaths = ["tests"]
 console_output_style = "progress-even-when-capture-no"
 markers = [
--- cryptography-50.0.0/tests/test_fernet.py
+++ cryptography-50.0.0/tests/test_fernet.py
@@ -10,7 +10,6 @@ import os
 import time
 import typing
 
-import pretend
 import pytest
 
 import cryptography_vectors
@@ -112,6 +111,7 @@ class TestFernet:
         with pytest.raises(TypeError):
             f.decrypt(typing.cast(typing.Any, 12345))
 
+    @pytest.mark.skip(reason="We don't have pretend on Solaris")
     def test_timestamp_ignored_no_ttl(self, monkeypatch):
         f = Fernet(base64.urlsafe_b64encode(b"\x00" * 32))
         pt = b"encrypt me"
--- cryptography-46.0.3/tests/test_utils.py
+++ cryptography-46.0.3/tests/test_utils.py
@@ -8,7 +8,6 @@ import inspect
 import os
 import textwrap
 
-import pretend
 import pytest
 
 import cryptography
@@ -46,6 +45,7 @@ def test_int_to_bytes_rejects_zero_lengt
         cryptography.utils.int_to_bytes(0, 0)
 
 
+@pytest.mark.skip(reason="We don't have pretend on Solaris")
 def test_check_backend_support_skip():
     supported = pretend.stub(
         kwargs={"only_if": lambda backend: False, "skip_message": "Nope"}
@@ -57,6 +57,7 @@ def test_check_backend_support_skip():
     assert exc_info.value.args[0] == "Nope (True)"
 
 
+@pytest.mark.skip(reason="We don't have pretend on Solaris")
 def test_check_backend_support_no_skip():
     supported = pretend.stub(
         kwargs={"only_if": lambda backend: True, "skip_message": "Nope"}
